### DISCUSSION TITLE
New config sidebar dont save selected values and overwrite system wide config.

### DISCUSSION THREAD

**Discussion by luisbrandao** [open-webui/open-webui#4094] [General]
# Bug Report

## Description

**Bug Summary:**
The new "sidebar" chat controls don't save the config, it is lost on page refresh.
If they are set to default, it get the web-ui default (for example, 2048 of context) and ignore system wide config:
![image](https://github.com/user-attachments/assets/c0f9e995-1e33-4d55-82a3-ddefb4d2d519)
![image](https://github.com/user-attachments/assets/9e59339c-4d40-473e-8416-c652a034d563)
![image](https://github.com/user-attachments/assets/95e55b29-5070-464f-b523-b67c5d9cd33a)

**Steps to Reproduce:**
Get in any chat, fiddle with the sidebar controls and then refresh.

**Expected Behavior:**
Default value either keep between refreshes or at least fall back to system default.

**Actual Behavior:**
Config are lost between refreshes.

## Environment

- **Open WebUI Version:** 0.3.10
- **Ollama (if applicable):** 0.2.8

- **Operating System:** [Rocky8/9]
- **Browser (if applicable):** [Brave browser]

## Logs and Screenshots

**Browser Console Logs:**
no log generated when config is fiddle with. There are somethings on reload. They dont prevent webui from working
![image](https://github.com/user-attachments/assets/c19f6ba6-ef7f-48e7-889d-4e1012f0e5ef)

**Docker Container Logs:**
no log generated when config is fiddle with. (no request sent from browser to webui)

## Installation Method
Docker with httpd reverse proxy and https.

**Comment by justinh-rahb** [comment]
It's not intended for these values to be saved, they're for the session you're currently in. As such it also makes sense for them to override the global values as determined by consensus in the original issue/PR. I have not observed that any parameters are sent at all if everything is `Default` everywhere.

All `Default`:
![Screenshot 2024-07-24 at 10 11 13 AM](https://github.com/user-attachments/assets/687ddeca-3567-4b93-99f5-a925a4729f79)

Setting `num_ctx`:
![Screenshot 2024-07-24 at 10 11 34 AM](https://github.com/user-attachments/assets/29924e4b-894b-4de7-88da-360d54e9153d)

**Reply by petergerten** [reply]
> It's not intended for these values to be saved, they're for the session you're currently in.

so for o1-preview I currently need to set streaming off and the system prompt to empty to make it work at all. What can I do to make these settings persistent?

**Reply by bsvaz** [reply]
> > It's not intended for these values to be saved, they're for the session you're currently in.
> 
> so for o1-preview I currently need to set streaming off and the system prompt to empty to make it work at all. What can I do to make these settings persistent?

Change in the settings.
![image](https://github.com/user-attachments/assets/5d119ad5-2a59-43ef-ad7e-66e32038c89e)

**Comment by luisbrandao** [comment]
I think there is something not working well while using the reverse proxy.
I just tried to connect using the direct container ip and the behavior is way more consistent.
